### ISSUE TITLE
Please help to add new device (check new device file)

### ISSUE THREAD

**Issue by Der-Nax** [make-all/tuya-local#192]
I have a new device power meter for 3 fases.
I try to make the new device file, but I did error. COuld you please check and show error.
```yaml
name: Power Clamp
primary_entity:
  entity: sensor
  class: energy
  dps:
    - id: 131
      type: energy
      name: total energy
      mapping:
        - scale: 100
      unit: kWh
      class: total_increasing
    - id: 132
      type: current
      name: total current
      mapping:
        - scale: 1000
      unit: A
    - id: 133
      type: power
      name: active power
      unit: W
    - id: 135
      type: frequency
      name: frequency
    - id: 136
      type: temperature
      name: temperature
      mapping:
        - scale: 10
secondary_entities:
  - entity: sensor
    category: diagnostic
    name: Voltage A
    dps:
      - id: 101
        name: sensor
        type: voltage
        unit: V
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: current
    name: Current A
    dps:
      - id: 102
        name: sensor
        type: current
        unit: A
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: power
    name: Power A
    dps:
      - id: 103
        name: sensor
        type: power
        unit: W
  - entity: sensor
    category: diagnostic
    class: power_factor
    name: Power factor A
    dps:
      - id: 104
        name: sensor
        type: power_factor
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: energy
    name: Energy A
    dps:
      - id: 106
        name: sensor
        type: energy
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    name: Voltage B
    dps:
      - id: 111
        name: sensor
        type: voltage
        unit: V
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: current
    name: Current B
    dps:
      - id: 112
        name: sensor
        type: current
        unit: A
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: power
    name: Power B
    dps:
      - id: 113
        name: sensor
        type: power
        unit: W
  - entity: sensor
    category: diagnostic
    class: power_factor
    name: Power factor B
    dps:
      - id: 114
        name: sensor
        type: power_factor
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: energy
    name: Energy B
    dps:
      - id: 116
        name: sensor
        type: energy
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    name: Voltage C
    dps:
      - id: 121
        name: sensor
        type: voltage
        unit: V
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: current
    name: Current C
    dps:
      - id: 122
        name: sensor
        type: current
        unit: A
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: power
    name: Power C
    dps:
      - id: 123
        name: sensor
        type: power
        unit: W
  - entity: sensor
    category: diagnostic
    class: power_factor
    name: Power factor C
    dps:
      - id: 124
        name: sensor
        type: power_factor
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: energy
    name: Energy C
    dps:
      - id: 126
        name: sensor
        type: energy
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100
```
About device:
I check it on localtuya and mapping all dps:
"model": "Power Clamp",
                            "dps_strings": [
                                "101 (value: 2284)",
                                "102 (value: 1073)",
                                "103 (value: 191)",
                                "104 (value: 78)",
                                "106 (value: 251)",
                                "111 (value: 2354)",
                                "112 (value: 748)",
                                "113 (value: 47)",
                                "114 (value: 100)",
                                "116 (value: 267)",
                                "121 (value: 2350)",
                                "122 (value: 753)",
                                "123 (value: 149)",
                                "124 (value: 84)",
                                "126 (value: 517)",
                                "131 (value: 1036)",
                                "132 (value: 2574)",
                                "133 (value: 188)",
                                "135 (value: 50)",
                                "136 (value: 390)"
                            ],
                            "entities": [
                                {
                                    "id": 101,
                                    "friendly_name": "F1_Voltage",
                                    "unit_of_measurement": "V",
                                    "device_class": "voltage",
                                    "scaling": 0.1,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 102,
                                    "friendly_name": "Current",
                                    "unit_of_measurement": "A",
                                    "device_class": "current",
                                    "scaling": 0.1,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 111,
                                    "friendly_name": "F2_Voltage",
                                    "unit_of_measurement": "V",
                                    "device_class": "voltage",
                                    "scaling": 0.1,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 121,
                                    "friendly_name": "F3_Voltage",
                                    "unit_of_measurement": "V",
                                    "device_class": "voltage",
                                    "scaling": 0.1,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 102,
                                    "friendly_name": "F1_Current",
                                    "unit_of_measurement": "A",
                                    "device_class": "current",
                                    "scaling": 0.001,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 112,
                                    "friendly_name": "F2_Current",
                                    "unit_of_measurement": "A",
                                    "device_class": "current",
                                    "scaling": 0.001,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 122,
                                    "friendly_name": "F3_Current",
                                    "unit_of_measurement": "A",
                                    "device_class": "current",
                                    "scaling": 0.001,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 103,
                                    "friendly_name": "F1_Power",
                                    "unit_of_measurement": "W",
                                    "device_class": "power",
                                    "platform": "sensor"
                                },
                                {
                                    "id": 104,
                                    "friendly_name": "F1_PowerFactor",
                                    "device_class": "power_factor",
                                    "scaling": 0.01,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 106,
                                    "friendly_name": "F1_Energy",
                                    "unit_of_measurement": "kWh",
                                    "device_class": "energy",
                                    "scaling": 0.01,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 113,
                                    "friendly_name": "F2_Power",
                                    "unit_of_measurement": "W",
                                    "device_class": "power",
                                    "platform": "sensor"
                                },
                                {
                                    "id": 114,
                                    "friendly_name": "F2_PowerFactor",
                                    "device_class": "power_factor",
                                    "scaling": 0.01,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 116,
                                    "friendly_name": "F2_Energy",
                                    "unit_of_measurement": "kWh",
                                    "device_class": "energy",
                                    "scaling": 0.01,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 123,
                                    "friendly_name": "F3_Power",
                                    "unit_of_measurement": "W",
                                    "device_class": "power",
                                    "platform": "sensor"
                                },
                                {
                                    "id": 124,
                                    "friendly_name": "F3_PowerFactor",
                                    "device_class": "power_factor",
                                    "scaling": 0.01,
                                    "platform": "sensor"
                                },
                                {
                                    "id": 126,
                                    "friendly_name": "F3_Energy",
                                    "unit_of_measurement": "kWh",
                                    "device_class": "energy",
                                    "scaling": 0.01,
                                    "platform": "sensor"
                                },
                              
                                }
                            ],
                            "product_key": "r64livttuufctble"

**Comment by make-all** [comment]
sensor entities must have a dps named "sensor". Your primary entity does not have one.  Also, dps names should not have spaces in them.

**Comment by Der-Nax** [comment]
I did it! You may add new device in your collection.
name: Power Clamp
primary_entity:
  entity: sensor
  class: energy
  dps:
    - id: 131
      type: integer
      name: sensor
      mapping:
        - scale: 100
      unit: kWh
      class: total_increasing
secondary_entities:
  - entity: sensor
    category: diagnostic
    name: Voltage A
    class: voltage
    dps:
      - id: 101
        name: sensor
        type: integer
        unit: V
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: current
    name: Current A
    dps:
      - id: 102
        name: sensor
        type: integer
        unit: A
        mapping:
          - scale: 1000
  - entity: sensor
    category: diagnostic
    class: power
    name: Power A
    dps:
      - id: 103
        name: sensor
        type: integer
        unit: W
  - entity: sensor
    category: diagnostic
    class: power_factor
    name: Power factor A
    dps:
      - id: 104
        name: sensor
        type: integer
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: energy
    name: Energy A
    dps:
      - id: 106
        name: sensor
        type: integer
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    name: Voltage B
    class: voltage
    dps:
      - id: 111
        name: sensor
        type: integer
        unit: V
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: current
    name: Current B
    dps:
      - id: 112
        name: sensor
        type: integer
        unit: A
        mapping:
          - scale: 1000
  - entity: sensor
    category: diagnostic
    class: power
    name: Power B
    dps:
      - id: 113
        name: sensor
        type: integer
        unit: W
  - entity: sensor
    category: diagnostic
    class: power_factor
    name: Power factor B
    dps:
      - id: 114
        name: sensor
        type: integer
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: energy
    name: Energy B
    dps:
      - id: 116
        name: sensor
        type: integer
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    name: Voltage C
    class: voltage
    dps:
      - id: 121
        name: sensor
        type: integer
        unit: V
        mapping:
          - scale: 10
  - entity: sensor
    category: diagnostic
    class: current
    name: Current C
    dps:
      - id: 122
        name: sensor
        type: integer
        unit: A
        mapping:
          - scale: 1000
  - entity: sensor
    category: diagnostic
    class: power
    name: Power C
    dps:
      - id: 123
        name: sensor
        type: integer
        unit: W
  - entity: sensor
    category: diagnostic
    class: power_factor
    name: Power factor C
    dps:
      - id: 124
        name: sensor
        type: integer
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: energy
    name: Energy C
    dps:
      - id: 126
        name: sensor
        type: integer
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100
  - entity: sensor
    category: diagnostic
    class: current
    name: Total current
    dps:
      - id: 132
        type: integer
        name: sensor
        mapping:
          - scale: 1000
        unit: A
  - entity: sensor
    category: diagnostic
    class: power
    name: Total active power
    dps:
      - id: 133
        type: integer
        name: sensor
        unit: W
  - entity: sensor
    category: diagnostic
    class: frequency
    name: Frequency
    dps:
      - id: 135
        type: integer
        name: sensor
  - entity: sensor
    category: diagnostic
    class: temperature
    name: Temperature
    dps:
      - id: 136
        type: integer
        name: sensor
        mapping:
          - scale: 10

Device (Tuya Smart Life Single 3 Phase WiFi Energy KWH Meter 300A 200A 80A Clamp Current Transformer App Monitor Energy Record Power):
![image](https://user-images.githubusercontent.com/108026974/178155542-1cb8da39-481d-4597-83f1-4ce345d423be.png)

**Comment by make-all** [comment]
Is the model number PC321-TY?  I can find several different brands with this model number matching the above picture (or differing only in the plastic casing - ie added vent holes or chamfered edges).

**Comment by Der-Nax** [comment]
I'll check User Manual later. At this moment I can provide information from Tuya-IOT - Get Device Information:
{
  "result": {
    "active_time": 1656860605,
    "category": "zndb",
    "category_name": "Smart Electric Meter",
    "create_time": 1656854981,
    "gateway_id": "",
    "icon": "smart/icon/ay1515026431018saCFs/272462f25ade846b58feb0f74916530a.png",
    "id": "xxx",
    "ip": "xxx",
    "lat": "xxx",
    "local_key": "xxx",
    "lon": "xxx",
    "model": "PC321",
    "name": "Power Clamp New House",
    "online": false,
    "owner_id": "xxx",
    "product_id": "r64livttuufctble",
    "product_name": "Power Clamp",
    "sub": false,
    "time_zone": "xxx",
    "update_time": xx,
    "uuid": "xxx"
  },
  "success": true,
  "t": xxx,
  "tid": "xxx"
It's look like you are right.

I make configuration file for one more device. How to transfer it for you? Inside this issue or open one more or may be any direct channel for you more friendly.

**Comment by make-all** [comment]
Open another issue for the device.  And please use three backticks to mark the start and end of the code block if you paste the yaml inline - otherwise there will be formatting changes.

**Comment by make-all** [comment]
This is included in the 0.18.0 release
